### PR TITLE
Refs [TASK][26642] Update RSpec for User and Campaign (using Devise)

### PR DESCRIPTION
**Pull request by hoatb-1022**
https://edu-redmine.sun-asterisk.vn/issues/26642

Screen Shot 2020-08-05 at 14 23 30

### CODE REVIEW

**Comment by hoatb-1022** [comment]
ready

File: spec/support/controller_macros.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module ControllerMacros
+  def log_in user
+    @request.env["devise.mapping"] = Devise.mappings[:user]
+    user = FactoryBot.create(:user) if user.nil?
```
Comment (ngocvt-0484) [inline spec/support/controller_macros.rb]:
```suggestion
    user ||= FactoryBot.create(:user) 
```

Comment (hoatb-1022) [reply spec/support/controller_macros.rb]: Oh man, nice, em quênn đấy :(

**Comment by hoatb-1022** [comment]
ready

**Pull request merged**